networkx/algorithms/minors.py

```diff
@@ -220,7 +220,27 @@ def quotient_graph(G, partition, edge_relation=None, node_data=None,
     # the blocks of the partition on the nodes of G induced by the
     # equivalence relation.
     if callable(partition):
+        # equivalence_classes always return partition of whole G.
         partition = equivalence_classes(G, partition)
+        return _quotient_graph(G, partition, edge_relation, node_data,
+                               edge_data, relabel, create_using)
+
+    # If the user provided partition as a collection of sets. Then we
+    # need to check if partition covers all of G nodes. If the answer
+    # is 'No' then we need to prepare suitable subgraph view.
+    partition_nodes = set().union(*partition)
+    if len(partition_nodes) != len(G):
+        if create_using is None:
+            # This hack is necessary because we will
+            # be passing subgraph instance.
+            create_using = G
```

Comment:
Instead of hacking create_using, could you fix the create_using treatment in _quotient_graph()?
We don't use ```type(create_using)()``` because it doesn't work with graph views (including subgraphs).

Please change line 247 in your file from

    H = type(create_using)() if create_using is not None else type(G)()

to 

    H = G.fresh_copy() if create_using is None else create_using.fresh_copy()

Thanks for helping me catch bugs!   :)